### DISCUSSION TITLE
[PSD3] Exercise 5.31

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#131] [Esercizi (A-L)]
**Exercise 5.31** Express the following base 10 numbers in 16-bit fixed-point two’s complement format with eight integer bits and eight fraction bits. Express your answer in hexadecimal.

(a) $−13.5625$
(b) $42.3125$
(c) $−17.15625$

- _**N.B.:** I numeri da convertire sono gli stessi dell'Esercizio 5.29 (#130 ) ma questa volta sono da esprimere in formato complemento a due anziché in formato sign/magnitude._

**Comment by Elia-Belli** [comment]
Converto da Decimale a Ca2 come se fosse un numero senza virgola, se è negativo ne faccio il complemento.

(a) $00001101.10010000 \rightarrow 11110010.01101111 + 1 = 11110010.01110000_2 = F270_{16}$
(b) $00101010.01010000_2 = 2A50_{16}$
(c) $00010001.00101000 \rightarrow 11101110.11010111 + 1 = 11101110.11011000_2 = EED8_{16}$

**Comment by FeddyLix17** [comment]
(a) -13.5625 (base 10) = 11110011.10010000 (Ca2) = F390 (base 16)
(b) 42.3125 (base 10) = 00101010.0101000 (Ca2) = 2A50 (base 16)
(c) -17.15625 (base 10) = 11101111.00101000 (Ca2) = DF28 (base 16)

**Comment by Elia-Belli** [comment]
La (a) e (c) di @FedVlogger17 sono errate, mentre la (b) è corretta.
C'erano degli errori nella mia (a) e (b) che sono stati corretti.

**Comment by FeddyLix17** [comment]
@Elia-Belli ho rifatto anche io i calcoli, e tutte le tue attuali soluzioni sono corrette. 
Mi sono andato a informare meglio su forum come  https://stackoverflow.com/questions/15463491/how-to-represent-a-negative-number-with-a-fraction-in-2s-complement e
https://math.stackexchange.com/questions/747669/binary-representation-of-a-fraction-in-twos-complement
per comprendere che non stessi facendo bene a priori il complemento a 2 dei numeri negativi, grazie per la correzione!
